java/arcs/android/storage/service/BindingContext.kt

```diff
@@ -57,7 +62,27 @@ class BindingContext(
     private val sendQueue = SendQueue()
 
     /** The local [CoroutineContext]. */
-    private val coroutineContext = parentCoroutineContext + CoroutineName("BindingContext-$id")
+    private val job = Job(parentCoroutineContext[Job])
+    private val coroutineContext =
+        parentCoroutineContext + job + CoroutineName("BindingContext-$id")
+
+    override fun idle(timeoutMillis: Long, resultCallback: IResultCallback) {
+        bindingContextStatisticsSink.traceTransaction("idle") {
+            bindingContextStatisticsSink.measure(coroutineContext + Dispatchers.IO) {
+                val activeStore = store.activate()
+                try {
+                    withTimeout(timeoutMillis) {
+                        activeStore.idle()
```

Comment:
Is there an issue that every time this is called, it creates a new ActiveStore?

Reply:
no, it only creates it if it needs to. If an active store has already been created from the base `Store`, then it just returns that one